File: LeetCode/C#/876. Middle of the Linked List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MiddleNode(ListNode head) {
        var current = head;
        var nodeCount = 0;
        
        while(current != null)
        {
            nodeCount++;
            current = current.next;
        }
        
        if(nodeCount == 1)
        {
            return head;
        }
        
        current = head;
        var middleIndex = nodeCount / 2;
        var currentIndex = 0;
        
        while(currentIndex != middleIndex)
        {
            current = current.next;
            currentIndex++;
        }
        
        return current;
        
    }
}

File: LeetCode/Java/876. Middle of the Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ArrayList<ListNode> nodeList = new ArrayList<ListNode>();

        while(head != null) {
            nodeList.add(head);
            head = head.next;
        }

        return nodeList.get(nodeList.size() / 2);
    }
}